.github/actions/run-android-device-farm-test/action.yml

```diff
@@ -0,0 +1,58 @@
+name: 'Run Android tests on Device Farm'
+inputs:
+  apk-path:
+    required: true
+  app-id:
+    required: true
+  project-arn:
+    required: true
+  device-pool-arn:
+    required: true
+  arguments:
+    default: _
+outputs:
+  test-results-path:
+    value: ${{ steps.get-test-results.outputs.results-path }}
+runs:
+  using: "composite"
+  steps:
+    - name: Run the tests
+      uses: realm/aws-devicefarm/test-application@d94e739490340474f8464f74289a0f48706dd3a3
+      id: run-tests
+      with:
+        project_arn: ${{ inputs.project-arn }}
+        device_pool_arn: ${{ inputs.device-pool-arn }}
+        app_file: ${{ inputs.apk-path }}
+        app_type: ANDROID_APP
+        test_type: APPIUM_PYTHON
+        test_package_file: https://github.com/realm/aws-devicefarm-sample-data/releases/download/0.3/sample_env_python3.zip
+        test_package_type: APPIUM_PYTHON_TEST_PACKAGE
+        test_spec_file: test_spec-${{ inputs.app-id }}.yaml
+        test_spec_type: APPIUM_PYTHON_TEST_SPEC
+        remote_src: true
+        test_spec: |
+          version: 0.1
+          phases:
+            install:
+              commands:
+                - export PYTHON_VERSION=3
+            test:
+              commands:
+                - adb shell am instrument -w -r io.realm.testapp.test/androidx.test.runner.AndroidJUnitRunner | egrep 'OK \([0-9]+ test[s]?\)'
+
+    - run: |
```

Comment:
How are failing tests being reported?

Reply:
The instrument run doesn't build the JUnit XML results to pull from the device. 
I modified the behaviour to always pull the Logcat from the device and publish it in case we want to investigate a failure/crash